toolbox/deps_update/main.go

```diff
@@ -129,54 +129,61 @@ func updateDependenciesOf(repo string) error {
 		log.Printf("Branch already exists")
 	}
 	// if branch exists, stop here and do not create another PR of identical delta
-	if err = githubClnt.CloseFailedPullRequests(
+	if err = githubClnt.CloseIdlePullRequests(
 		prTitlePrefix, repo, *baseBranch); exists || err != nil {
 		return err
 	}
-	if _, err := u.Shell("git checkout -b " + branch); err != nil {
+	if _, err = u.Shell("git checkout -b " + branch); err != nil {
 		return err
 	}
-	if err := updateDeps(repo, &deps, &depChangeList); err != nil {
+	if err = updateDeps(repo, &deps, &depChangeList); err != nil {
 		return err
 	}
-	if err := u.SerializeDeps(istioDepsFile, &deps); err != nil {
+	if err = u.SerializeDeps(istioDepsFile, &deps); err != nil {
 		return err
 	}
-	if err := u.CreateCommitPushToRemote(branch, "Update_Dependencies"); err != nil {
+	if err = u.CreateCommitPushToRemote(branch, "Update_Dependencies"); err != nil {
 		return err
 	}
 	prTitle := prTitlePrefix + repo
-	return githubClnt.CreatePullRequest(prTitle, prBody, branch, *baseBranch, repo)
+	pr, err := githubClnt.CreatePullRequest(prTitle, prBody, branch, *baseBranch, repo)
+	if err != nil {
+		return err
+	}
+	return githubClnt.AddlLGMTandApprovedLabelsToPR(repo, pr)
 }
 
-func main() {
+func init() {
 	flag.Parse()
 	if *tokenFile == "" {
-		log.Panicf("token_file not provided\n")
+		log.Fatalf("token_file not provided\n")
 		return
 	}
 	token, err := u.GetAPITokenFromFile(*tokenFile)
 	if err != nil {
-		log.Panicf("Error accessing user supplied token_file: %v\n", err)
+		log.Fatalf("Error accessing user supplied token_file: %v\n", err)
 	}
 	githubClnt = u.NewGithubClient(*owner, token)
+}
+
+func main() {
 	if *repo != "" { // only update dependencies of this repo
 		if (*repo == "istio") == (*hub == "") {
-			log.Printf("The hub flag hub must be set for istio/istio and must not for other repos\n")
+			log.Fatalf("The hub flag hub must be set for istio/istio and must not for other repos\n")
 			return
 		}
 		if err := updateDependenciesOf(*repo); err != nil {
-			log.Printf("Failed to udpate dependency: %v\n", err)
+			log.Fatalf("Failed to udpate dependency: %v\n", err)
```

Comment:
I don't understand why changed back to fatalf? @chxchx

Reply:
Because in calling fatal will have the binary return a nonzero status code to the caller, which is useful in prow so that prow will mark this particular run as failed rather than silently failing.